test/integration/text_layer_spec.mjs

```diff
@@ -98,111 +99,270 @@ describe("Text layer", () => {
+              await page.mouse.up();
+
+              await expectAsync(page)
+                .withContext(`In ${browserName}`)
+                .toHaveRoughlySelected("a.org\nTe");
+            })
+          );
+        });
+
+        it("allows clicking the link after selecting", async () => {
+          await Promise.all(
+            pages.map(async ([browserName, page]) => {
+              const [positionStart, positionEnd] = await Promise.all([
+                getSpanRectFromText(page, 1, "Link").then(middleLeftPosition),
+                getSpanRectFromText(page, 1, "mozilla.org").then(
+                  middlePosition
+                ),
+              ]);
+
+              await page.mouse.move(positionStart.x, positionStart.y);
+              await page.mouse.down();
+              await moveInSteps(page, positionStart, positionEnd, 20);
+              await page.mouse.up();
+
+              const clickPromiseHandle = await waitForClick(
+                page,
+                "#pdfjs_internal_id_8R",
+                1000
               );
 
-            await moveInSteps(page, positionStartPage2, positionEndPage2, 20);
-            await page.mouse.up();
+              await page.mouse.click(positionEnd.x, positionEnd.y);
 
-            await expectAsync(page)
-              .withContext(`In ${browserName}, second selection`)
-              .toHaveRoughlySelected(
-                /path through.*Hence, recording and .* tracing, and give/s
+              const clicked = await awaitPromise(clickPromiseHandle);
+              expect(clicked).toBeTrue();
+            })
+          );
+        });
+
+        it("allows clicking the link after changing selection with the keyboard", async () => {
+          await Promise.all(
+            pages.map(async ([browserName, page]) => {
+              const [positionStart, positionEnd] = await Promise.all([
+                getSpanRectFromText(page, 1, "Link").then(middleLeftPosition),
+                getSpanRectFromText(page, 1, "mozilla.org").then(
+                  middlePosition
+                ),
+              ]);
+
+              await page.mouse.move(positionStart.x, positionStart.y);
+              await page.mouse.down();
+              await moveInSteps(page, positionStart, positionEnd, 20);
+              await page.mouse.up();
+
+              await page.keyboard.down("Shift");
+              await page.keyboard.press("ArrowRight");
+              await page.keyboard.up("Shift");
+
+              const clickPromiseHandle = await waitForClick(
+                page,
+                "#pdfjs_internal_id_8R",
+                1000
               );
-          })
-        );
+
+              await page.mouse.click(positionEnd.x, positionEnd.y);
+
+              const clicked = await awaitPromise(clickPromiseHandle);
+              expect(clicked).toBeTrue();
+            })
+          );
+        });
+
+        function waitForClick(page, selector, timeout) {
```

Comment:
Let's move this helper function to the top of the `describe` block so it's defined before using it (we generally prefer this in the tests).